v0/destinations/ga4/transform.js

```diff
@@ -34,6 +34,32 @@ const {
   GA4_PARAMETERS_EXCLUSION
 } = require("./utils");
 
+/**
+ * Returns GA4 client_id
+ * @param {*} message
+ * @param {*} clientIdFieldIdentifier
+ * @returns
+ */
+const getGA4ClientId = (message, { clientIdFieldIdentifier }) => {
+  // first we will search from webapp
+  if (clientIdFieldIdentifier) {
+    return get(message, clientIdFieldIdentifier);
```

Comment:
Do we need to fallback to default values if the value is not found here? If not we need to throw proper error here

Reply:
@akashrpo yes, we need to fall back to default search since `clientIdFieldIdentifier` is optional from webapp.

Reply:
well we have a return statement here in that case it will throw error instead

Reply:
Let's add a test case for this too

Reply:
updated

Reply:
we are not yet taking the values from the fallback types

Reply:
updated